File: src/main/java/no/nav/syfo/controller/domain/Kontaktinfo.java
package no.nav.syfo.controller.domain;


import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;
import java.io.Serializable;

@Data
@Accessors(fluent = true, chain = true)
@EqualsAndHashCode
public class Kontaktinfo implements Serializable {

    public String fnr;
    public String epost;
    public String tlf;
    public Boolean skalHaVarsel;
    public FeilAarsak feilAarsak;


    public enum FeilAarsak {
        RESERVERT,
        UTGAATT,
        KONTAKTINFO_IKKE_FUNNET,
        SIKKERHETSBEGRENSNING,
        PERSON_IKKE_FUNNET
    }
}

File: src/main/kotlin/no/nav/syfo/controller/domain/Kontaktinfo.kt
package no.nav.syfo.controller.domain

import java.io.Serializable

data class Kontaktinfo(
    val fnr: String? = null,
    val epost: String? = null,
    val tlf: String? = null,
    val skalHaVarsel: Boolean? = null,
    val feilAarsak: FeilAarsak? = null
) : Serializable

enum class FeilAarsak {
    RESERVERT,
    UTGAATT,
    KONTAKTINFO_IKKE_FUNNET,
    SIKKERHETSBEGRENSNING,
    PERSON_IKKE_FUNNET
}
